src/language.rs

```diff
@@ -28,11 +28,11 @@ use thiserror::Error;
 ///
 /// See [`SymbolLang`] for quick-and-dirty use cases.
 #[allow(clippy::len_without_is_empty)]
-pub trait Language: Debug + Clone + Eq + Ord + Hash {
+pub trait Language: Debug + Clone + Eq + Ord + Hash + Send + Sync {
```

Comment:
I don't think it's acceptable to change the bounds of `Language` at this point in `egg`'s lifecycle. Getting rid of this I think is the main challenge the merging this.

Reply:
I added a new trait `MaybePar`, which is
- automatically implemented for every type when `parallel-matching` is not set,
- automatically implemented for `Sync + Send` types when `parallel-matching` is not set.
`Language`, `Analysis`, `Analysis::Data`, and some other traits now have bounds involving this trait, so nothing changes when `parallel-matching` isn't set. Let me know what you think about this solution. If [trait aliases](https://doc.rust-lang.org/unstable-book/language-features/trait-alias.html) ever get stabilized, they could be used instead.